ci: fix reference to android job filename in main.yml

mukilan opened the pull request:
<!-- Please describe your changes on the following line: -->


---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because the changes fix the android CI job added recently

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

mukilan committed:
ci: fix reference to android job filename in main.yml

atbrakhi commented on .github/workflows/main.yml:
```diff
@@ -129,7 +129,7 @@ jobs:
     needs: ["decision"]
     if: ${{ contains(fromJson(needs.decision.outputs.configuration).platforms, 'android') }}
-    uses: ./.github/workflows/linux.yml
+    uses: ./.github/workflows/android.yml
```
We have a `android.yml` file [already](https://github.com/servo/servo/blob/main/.github/workflows/android.yml)

mukilan committed:
mach: fix build command to handle android

The refactoring done in PR #31092 introduced the call to
get_binary_path in the unconditional path, but did not
pass the android flag to the call.

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

mukilan commented:
@mrobinson looks like the build breaks because of the call to get_binary_path added in #31092. I've pushed a fix in the latest commit to address this. Please review and let me know if you see a better solution.

mrobinson commented:
> @mrobinson looks like the build breaks because of the call to get_binary_path added in #31092. I've pushed a fix in the latest commit to address this. Please review and let me know if you see a better solution.

It looks good. Thanks for the fix!
